scc/src/main.rs

```diff
@@ -37,45 +39,69 @@ fn main() -> ExitCode {
             .as_slice(),
     );
 
-    #[cfg(test)]
-    log::info!("{:#?}", opts);
+    let scripts_dir = if let Some(dir) = opts.scripts_dir {
+        dir
+    } else {
+        println!("Error: r6/scripts directory is required");
+        return ExitCode::FAILURE;
+    };
 
-    let script_dir = PathBuf::from(opts.script_paths.first().unwrap());
-    let r6_dir = script_dir.parent().unwrap();
+    let r6_dir = scripts_dir.parent().unwrap();
     let default_cache_dir = r6_dir.join("cache");
 
     setup_logger(r6_dir);
 
-    let manifest = ScriptManifest::load(&script_dir).unwrap_or_else(|err| {
+    let manifest = ScriptManifest::load(&scripts_dir).unwrap_or_else(|err| {
         log::info!("Using defaults for the script manifest ({err})");
         ScriptManifest::default()
     });
 
-    let mut script_paths = opts.script_paths;
+    let mut script_paths = vec![scripts_dir.clone()];
     match opts.script_paths_file.as_deref().map(load_script_paths).transpose() {
         Ok(loaded_paths) => script_paths.extend(loaded_paths.unwrap_or_default()),
         Err(err) => log::warn!("An invalid script paths file was provided: {err}, it will be ignored"),
     };
 
-    let (cache_dir, fallback_dir) = match opts.cache_dir {
-        Some(cache_dir) => {
-            log::info!("Custom cache directory provided: {}", cache_dir.to_str().unwrap());
-            let expected_bundle_path = cache_dir.join(BUNDLE_FILE_NAME);
-            if !expected_bundle_path.exists() {
-                let base = get_base_bundle_path(&default_cache_dir);
-                fs::create_dir_all(&cache_dir).expect("Could not create the custom cache directory");
-                fs::copy(base, expected_bundle_path).expect("Could not copy base bundle");
+    let (bundle_path, cache_dir, fallback_dir) = match opts.cache_file.as_deref() {
+        Some(file) => {
+            log::info!("Bundle path provided: {}", file.to_str().unwrap());
+            if opts.cache_dir.is_some() {
+                log::warn!("Custom cache directory also provided - ignoring");
             }
-            (cache_dir, Some(default_cache_dir))
+            (
+                file.to_path_buf(),
+                file.parent().unwrap().to_path_buf(),
+                Some(default_cache_dir.clone()),
+            )
         }
-        _ => (default_cache_dir, None),
+        _ => match opts.cache_dir.as_deref() {
```

Comment:
it might be a good idea to unnest the two pattern matches, by combining the two into a tuple:
```rs
match (opts.cache_file.as_deref(), opts.cache_dir.as_deref()) {
  (Some(cache_file), _) => ...
  (_, Some(cache_dir) => ...
  (_, _) => ...
}
```

Reply:
Ah, cool!